### PR TITLE
Check user exist - login -cookies!

### PR DESCRIPTION
**Pull request by antoniotrkdz**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -82,31 +101,113 @@ server.register(plugins, err => {
...
+        },
     });
 
     server.route({
         method: 'POST',
-        path: '/api/jobs',
+        path: '/api/reg-student',
         handler: (request, reply) => {
-            data.postJobs(request.payload, (err, res) => {
+            console.log(request.payload);
```
Comment (des-des) [inline server/index.js]: no

File: server/index.js [hunk 2]
```diff
@@ -82,31 +101,113 @@ server.register(plugins, err => {
...
+            hashPassword(request.payload.password, (err, hash) => {
                 if (err) {
-                    reply.status(500)(
-                        'Failed to connect load data from the database'
+                    return reply(Boom.badData('', 'bcrypt error'));
+                }
+                console.log(hash);
```
Comment (des-des) [inline server/index.js]: no

File: server/index.js [hunk 3]
```diff
@@ -58,20 +60,37 @@ server.register(plugins, err => {
...
+            data.postJobs(request.payload, (err, res) => {
+                if (err) {
+                    return reply.status(500)(
+                        'Failed to connect load data from the database'
+                    );
+                }
+                reply({
+                    name: 'newJob',
+                    message: 'Welcome to BEEVR!',
+                    newJob: res,
```
Comment (des-des) [inline server/index.js]: What is in res?

Is there anything sensitive in there? You should not really pass the db response directly to client

File: server/index.js [hunk 4]
```diff
@@ -82,31 +101,113 @@ server.register(plugins, err => {
                     reply({
                         name: 'studentList',
                         message: 'Welcome to BEEVR!',
-                        studentList: res
+                        studentList: res,
```
Comment (des-des) [inline server/index.js]: does this contain hashed password + emails and other sensitive information?

File: server/index.js [hunk 5]
```diff
@@ -82,31 +101,113 @@ server.register(plugins, err => {
...
             }, request.url.query.searchTerm);
-        }
+        },
+    });
+
+    server.route({
+        method: 'GET',
+        path: '/api/student',
+        handler: (request, reply) => {
+            console.log('req', request.query);
```
Comment (des-des) [inline server/index.js]: :fire:

File: server/index.js [hunk 6]
```diff
@@ -82,31 +101,113 @@ server.register(plugins, err => {
...
             });
-        }
+        },
+    });
+
+    server.route({
+        method: 'POST',
+        path: '/api/reg-resident',
+        handler: (request, reply) => {
+            console.log(request.payload);
```
Comment (des-des) [inline server/index.js]: please stop doing this

File: server/index.js [hunk 7]
```diff
@@ -82,31 +101,113 @@ server.register(plugins, err => {
...
+            hashPassword(request.payload.password, (err, hash) => {
+                if (err) {
+                    return reply(Boom.badData('', 'bcrypt error'));
+                }
+                console.log(hash);
```
Comment (des-des) [inline server/index.js]: :scream:

File: server/index.js [hunk 8]
```diff
@@ -82,31 +101,113 @@ server.register(plugins, err => {
...
+                data.postResidents(
+                    Object.assign({}, request.payload, {password_hash: hash}),
+                    (err, res) => {
+                        console.log(request.payload);
```
Comment (des-des) [inline server/index.js]: :-1:

File: server/index.js [hunk 9]
```diff
@@ -82,31 +101,113 @@ server.register(plugins, err => {
...
+                            return reply(
+                                Boom.serverUnavailable(
+                                    'unavailable',
+                                    data.error
+                                )
+                            );
+                        }
+                        reply({
+                            name: 'resident',
+                            resident: res,
```
Comment (des-des) [inline server/index.js]: what is in this response?

File: server/index.js [hunk 10]
```diff
@@ -124,11 +225,57 @@ server.register(plugins, err => {
...
+            const email = request.query.email;
+            data.loginRequest(email, (err, res) => {
+                if (err) {
+                    return reply(
+                        Boom.unauthorized(
+                            'Please log-in to see that',
+                            data.error
+                        )
+                    );
+                    console.log('ffff', err);
```
Comment (des-des) [inline server/index.js]: f*ck

File: server/index.js [hunk 11]
```diff
@@ -124,11 +225,57 @@ server.register(plugins, err => {
...
             });
-        }
+        },
+    });
+
+    server.route({
+        method: 'GET',
+        path: '/api/auth',
+        handler: (request, reply) => {
+            console.log(request.query);
```
Comment (des-des) [inline server/index.js]: ?

File: server/index.js [hunk 12]
```diff
@@ -124,11 +225,57 @@ server.register(plugins, err => {
...
+                    return reply(
+                        Boom.unauthorized(
+                            'Please log-in to see that',
+                            data.error
+                        )
+                    );
+                    console.log('ffff', err);
+                }
+                const user = res;
+                console.log('hey', user);
```
Comment (des-des) [inline server/index.js]: ...

File: server/index.js [hunk 13]
```diff
@@ -124,11 +225,57 @@ server.register(plugins, err => {
...
+                    (err, match) => {
+                        if (err) {
+                            return reply(
+                                Boom.unauthorized(
+                                    'Please log-in to see that',
+                                    data.error
+                                )
+                            );
+                        }
+                        console.log('passwords matched');
```
Comment (des-des) [inline server/index.js]: okay

File: react-ui/src/actions/login.js [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+        type: SET_AUTH,
+        status: 'pending'
+    });
+
+    axios
+        .get('/api/auth', {params: {email, password}})
+        .then(response => {
+            dispatch({
+                type: SET_AUTH,
+                status: 'succes',
```
Comment (antoniotrkdz) [inline react-ui/src/actions/login.js]: done.
deleted comments as well.

File: react-ui/src/actions/login.js [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+        .catch(error => {
+            dispatch({
+                type: SET_AUTH,
+                status: 'error',
+                error: 'Invalid credentials'
+            });
+        });
+};
+
+        //const  = res.rows[0];
```
Comment (antoniotrkdz) [inline react-ui/src/actions/login.js]: done

File: react-ui/src/actions/login.js [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+
+    axios
+        .get('/api/auth', {params: {email, password}})
+        .then(response => {
+            dispatch({
+                type: SET_AUTH,
+                status: 'succes',
+                response: response.confirm
+            })
+                .then(console.log(response))
```
Comment (antoniotrkdz) [inline react-ui/src/actions/login.js]: Done.

File: react-ui/src/components/Form_Register_Resident.js [hunk 17]
```diff
@@ -1,41 +1,59 @@
...
 
         this.state = {
-            resident: {}
+            resident: {},
+            errorMessage: '',
         };
     }
 
+    checkEmail(value) {
+        console.log(this.props);
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Resident.js]: Done.

File: react-ui/src/components/Form_Register_Resident.js [hunk 18]
```diff
@@ -1,41 +1,59 @@
...
+        console.log(this.state);
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Resident.js]: Done.

File: react-ui/src/components/Form_Register_Resident.js [hunk 19]
```diff
@@ -1,41 +1,59 @@
...
+        value = this.state.resident.email;
+        if (value !== '') {
+            this.props.checkIfResidentExists(value);
+            console.log(value);
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Resident.js]: Done.

Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Resident.js]: Done.

File: react-ui/src/components/Form_Register_Student.js [hunk 20]
```diff
@@ -1,38 +1,80 @@
...
         this.state = {
-            student: {}
+            student: {},
+            errorMessage: '',
+            isAuthenticated: false,
         };
     }
 
+    checkEmail(value) {
+        console.log(this.props);
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

File: react-ui/src/components/Form_Register_Student.js [hunk 21]
```diff
@@ -1,38 +1,80 @@
...
+        console.log(this.state);
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

File: react-ui/src/components/Form_Register_Student.js [hunk 22]
```diff
@@ -1,38 +1,80 @@
...
+        value = this.state.student.email;
+        if (value !== '') {
+            this.props.checkIfStudentExists(value);
+            console.log(value);
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

File: react-ui/src/components/Login_Form.js [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+                    <button className="btn btn-primary" type="submit">
+                        {this.props.btnText}
+                    </button>
+                </div>
+            </form>
+        );
+    }
+}
+
+//LoginForm.propTypes = {
```
Comment (antoniotrkdz) [inline react-ui/src/components/Login_Form.js]: Done.

File: react-ui/src/components/pages/RegisterResident.js [hunk 24]
```diff
@@ -2,26 +2,36 @@ import React, {Component} from 'react';
 import Form_Register_Resident from '../Form_Register_Resident.js';
 import LoadingIndicator from '../LoadingIndicator.js';
 import {Router, Route, IndexRoute, browseHistory} from 'react-router';
+import * as actions from '../../actions/register_resident.js';
+import {connect} from 'react-redux';
 
 class RegisterResident extends Component {
     render() {
-        const dispatch = this.props.dispatch;
+        //const dispatch = this.props.dispatch;
```
Comment (antoniotrkdz) [inline react-ui/src/components/pages/RegisterResident.js]: Done,

File: react-ui/src/components/pages/RegisterResident.js [hunk 25]
```diff
@@ -2,26 +2,36 @@ import React, {Component} from 'react';
...
         //const {formState, currentlySending} = this.props.data;
```
Comment (antoniotrkdz) [inline react-ui/src/components/pages/RegisterResident.js]: Done>

File: react-ui/src/components/pages/RegisterResident.js [hunk 26]
```diff
@@ -2,26 +2,36 @@ import React, {Component} from 'react';
...
         );
     }
 }
 
-export default RegisterResident; 
-    // Register a user
-    //_register(username, password) {
-    //this.props.dispatch(_register(username, password));
-    //}
+// Register a user
```
Comment (antoniotrkdz) [inline react-ui/src/components/pages/RegisterResident.js]: Done.

File: react-ui/src/components/Nav.js [hunk 27]
```diff
@@ -59,9 +51,9 @@ class Nav extends Component {
     }
 }
 
-Nav.propTypes = {
-    loggedIn: React.PropTypes.bool.isRequired,
-    currentlySending: React.PropTypes.bool.isRequired
-};
+//Nav.propTypes = {
```
Comment (antoniotrkdz) [inline react-ui/src/components/Nav.js]: Done.

Comment (antoniotrkdz) [inline react-ui/src/components/Nav.js]: Time constraints....

File: react-ui/src/components/pages/RegisterStudent.js [hunk 28]
```diff
@@ -2,8 +2,10 @@ import React, {Component} from 'react';
 import Form_Register_Student from '../Form_Register_Student.js';
 import LoadingIndicator from '../LoadingIndicator.js';
 import {Router, Route, IndexRoute, browseHistory} from 'react-router';
+import * as actions from '../../actions/register_student.js';
+import {connect} from 'react-redux';
 
-class RegisterPage extends Component {
+class RegisterStudent extends Component {
     render() {
         //const dispatch = this.props.dispatch;
```
Comment (antoniotrkdz) [inline react-ui/src/components/pages/RegisterStudent.js]: Done.

Comment (antoniotrkdz) [inline react-ui/src/components/pages/RegisterStudent.js]: Done.

File: react-ui/src/components/pages/RegisterStudent.js [hunk 29]
```diff
@@ -27,13 +31,18 @@ class RegisterPage extends Component {
 }
 
 //function select(state) {
-    //return {data: state};
+//return {data: state};
```
Comment (antoniotrkdz) [inline react-ui/src/components/pages/RegisterStudent.js]: You did it...

File: react-ui/src/components/pages/RegisterStudent.js [hunk 30]
```diff
@@ -27,13 +31,18 @@ class RegisterPage extends Component {
...
 //}
 
-    //onSubmit: React.PropTypes.func.isRequired,
-    //btnText: React.PropTypes.string.isRequired,
-    //data: React.PropTypes.object.isRequired
+//onSubmit: React.PropTypes.func.isRequired,
```
Comment (antoniotrkdz) [inline react-ui/src/components/pages/RegisterStudent.js]: Done.

File: react-ui/src/components/pages/RegisterStudent.js [hunk 31]
```diff
@@ -27,13 +31,18 @@ class RegisterPage extends Component {
...
+//btnText: React.PropTypes.string.isRequired,
```
Comment (antoniotrkdz) [inline react-ui/src/components/pages/RegisterStudent.js]: Done.

File: react-ui/src/components/Login_Form.js [hunk 32]
```diff
@@ -0,0 +1,101 @@
...
+        this.setState({errorMessage: ''}, () => {
+            this.props.loginRequest(this.state.email, this.state.password);
+        });
+        (err) => this.setState({
+            errorMessage: err.data.error
+        });
+        console.log('hjhjh',this.state.errorMessage);
+    };
+
+    //onSubmit(evt) {
```
Comment (antoniotrkdz) [inline react-ui/src/components/Login_Form.js]: Done.

File: react-ui/src/components/Login_Form.js [hunk 33]
```diff
@@ -0,0 +1,101 @@
...
+        this.setState({
+            [evt.target.name]: evt.target.value,
+        });
+    }
+
+    render() {
+        console.log(this.state);
+        return (
+            <form className="form" onSubmit={this.onSubmit}>
+                <ErrorMessage />
```
Comment (antoniotrkdz) [inline react-ui/src/components/Login_Form.js]: old code....

File: react-ui/src/components/pages/RegisterStudent.js [hunk 34]
```diff
@@ -27,13 +31,18 @@ class RegisterPage extends Component {
...
+//data: React.PropTypes.object.isRequired
```
Comment (antoniotrkdz) [inline react-ui/src/components/pages/RegisterStudent.js]: Done.

File: server/index.js (hunk 1 above)
Comment (antoniotrkdz) [inline server/index.js]: Done.

File: server/index.js (hunk 5 above)
Comment (antoniotrkdz) [inline server/index.js]: Done.

File: server/index.js [hunk 35]
```diff
@@ -58,20 +60,37 @@ server.register(plugins, err => {
...
                         'Failed to connect load data from the database'
                     );
-                else {
-                    reply({
-                        name: 'jobsList',
-                        message: 'Welcome to BEEVR!',
-                        jobsList: res
-                    });
-                }
+                reply({
```
Comment (antoniotrkdz) [inline server/index.js]: corrected, forgot to return the error to exit the function.

File: server/index.js (hunk 2 above)
Comment (antoniotrkdz) [inline server/index.js]: Done.

File: server/index.js (hunk 6 above)
Comment (antoniotrkdz) [inline server/index.js]: Done.

File: server/index.js (hunk 7 above)
Comment (antoniotrkdz) [inline server/index.js]: Done.

File: server/index.js (hunk 11 above)
Comment (antoniotrkdz) [inline server/index.js]: Done.

File: server/index.js (hunk 10 above)
Comment (antoniotrkdz) [inline server/index.js]: Done.

File: server/index.js (hunk 12 above)
Comment (antoniotrkdz) [inline server/index.js]: Done

File: server/index.js (hunk 13 above)
Comment (antoniotrkdz) [inline server/index.js]: Removed.

File: react-ui/src/actions/register_resident.js [hunk 36]
```diff
@@ -20,7 +21,32 @@ export const registerResident = resident => dispatch => {
             dispatch({
                 type: REGISTER_RESIDENT,
                 status: 'error',
-                error: error
+                //error: error
```
Comment (antoniotrkdz) [inline react-ui/src/actions/register_resident.js]: Done.

File: react-ui/src/components/ErrorMessage.js [hunk 37]
```diff
@@ -11,9 +11,9 @@ let ErrorMessage = props => {
         : <div />;
 };
 
-ErrorMessage.propTypes = {
```
Comment (antoniotrkdz) [inline react-ui/src/components/ErrorMessage.js]: Deleted all validations. Time constraints.

File: react-ui/src/components/Form_Register_Resident.js [hunk 38]
```diff
@@ -180,7 +196,7 @@ class Form_Register_Resident extends Component {
                         onChange={this.onChange}
                     />
                 </div>
-                <div className="form__submit-btn-wrapper">
+                <div>
                     {this.props.currentlySending
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Resident.js]: I am leaving it as we can implement a spinner later.

File: react-ui/src/components/Form_Register_Student.js [hunk 39]
```diff
@@ -1,38 +1,80 @@
...
+    }
+
     onSubmit(evt) {
         evt.preventDefault();
-        this.props.registerStudent(this.state.student);
+        var student = this.state.student;
+        if (!student.email) {
+            var error_message = 'Email cannot be empty';
+        }
+        //var catArray = Object.keys(student).map((value) => {
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Removed.

File: react-ui/src/components/Form_Register_Student.js [hunk 40]
```diff
@@ -1,38 +1,80 @@
...
+            //if (value.indexOf('index_') !== -1) {
+                //return student[value];
+            //}
+        //}).filter((v) => {
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: removed.

File: react-ui/src/components/Form_Register_Student.js [hunk 41]
```diff
@@ -1,38 +1,80 @@
...
+            //return v !== undefined;
+        //});
+        //console.log('job-cat', catArray);
+        //Object.assign({}, student, {job_cat:catArray})
+        this.setState({errorMessage: error_message}, () => {
+            if (!this.state.errorMessage) {
+                this.props.registerStudent(student);
+                //browserHistory.push('/dashboard');
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: I am Leaving it as we might finally work out the react routing.

File: react-ui/src/components/Form_Register_Student.js [hunk 42]
```diff
@@ -1,38 +1,80 @@
...
                 ...student,
-                [evt.target.name]: evt.target.value
-            }
+                [evt.target.name]: evt.target.value,
+            },
         });
     }
 
     render() {
+        console.log(this.state);
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Leave the state logs as I need them while working on the form.

File: react-ui/src/components/Form_Register_Student.js [hunk 43]
```diff
@@ -1,38 +1,80 @@
...
+        //Object.assign({}, student, {job_cat:catArray})
+        this.setState({errorMessage: error_message}, () => {
+            if (!this.state.errorMessage) {
+                this.props.registerStudent(student);
+                //browserHistory.push('/dashboard');
+            }
+        });
+        (err) => {
+            this.setState({errorMessage: err.response.data.message});
+            console.log(err);
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: I know it is not doing it, but I would need something like this for the validation.
Please leave it as it doesn't break anything.

File: react-ui/src/components/Form_Register_Student.js [hunk 44]
```diff
@@ -1,38 +1,80 @@
...
+                [evt.target.name]: evt.target.value,
+            },
         });
     }
 
     render() {
+        console.log(this.state);
         return (
-            <form onSubmit={this.onSubmit}>
+            <form className="form" onSubmit={this.onSubmit}>
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: I am working on it....

File: react-ui/src/components/Form_Register_Student.js [hunk 45]
```diff
@@ -1,38 +1,80 @@
...
+                <div
+                    className="form-group"
+                    className={
+                        this.state.errorMessage ? 'alert alert-danger' : ''
+                    }
+                >
+                    {this.state.errorMessage}
+                </div>
                 <ErrorMessage />
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Sorry, as above leave it as I will work on the validation.

File: react-ui/src/components/Form_Register_Student.js [hunk 46]
```diff
@@ -189,69 +228,92 @@ class Form_Register_Student extends Component {
...
                     </label>
                     <input
                         className="form-control"
-                        name="index1"
-                        id="job category1"
+                        name="index_1"
+                        id="job_category_1"
                         type="text"
                         placeholder="Select"
+                        value={this.state.student.index_1}
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done in the form-multiselect branch.

File: react-ui/src/components/Form_Register_Student.js [hunk 47]
```diff
@@ -270,10 +332,10 @@ class Form_Register_Student extends Component {
                     </datalist>
                 </div>
 
-                <div className="form__submit-btn-wrapper">
+                <div>
                     {this.props.currentlySending
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: As per comment above.
Please leave while I try to implement a spinner.

File: react-ui/src/components/Login_Form.js [hunk 48]
```diff
@@ -0,0 +1,101 @@
...
+    onSubmit(evt) {
+        evt.preventDefault();
+        var state = this.props.state;
+        this.setState({errorMessage: ''}, () => {
+            this.props.loginRequest(this.state.email, this.state.password);
+        });
+        (err) => this.setState({
+            errorMessage: err.data.error
+        });
+        console.log('hjhjh',this.state.errorMessage);
```
Comment (antoniotrkdz) [inline react-ui/src/components/Login_Form.js]: removed.

File: react-ui/src/components/Login_Form.js [hunk 49]
```diff
@@ -0,0 +1,101 @@
...
+    onChange(evt) {
+        var email = this.state.email;
+        var password = this.state.password;
+        this.setState({
+            [evt.target.name]: evt.target.value,
+        });
+    }
+
+    render() {
+        console.log(this.state);
```
Comment (antoniotrkdz) [inline react-ui/src/components/Login_Form.js]: As above.

File: server/database/db_queries.js [hunk 50]
```diff
@@ -81,20 +122,37 @@ data.postStudents = (student, callback) => {
...
+    dbConnection.query(
+        `SELECT exists(
+            SELECT 1 FROM residents
+                WHERE email = $1);`,
+        [email],
+        (err, res) => {
+            if (err) {
+                callback(err);
+            }
+            callback(null, res.rows[0]);
```
Comment (antoniotrkdz) [inline server/database/db_queries.js]: As above.

File: react-ui/src/actions/login.js [hunk 51]
```diff
@@ -0,0 +1,43 @@
...
+import {SET_AUTH} from '../constants/action_types.js';
+
+export const loginRequest = (email, password) => dispatch => {
+    dispatch({
+        type: SET_AUTH,
+        status: 'pending'
+    });
+
+    axios
+        .get('/api/auth', {params: {email, password}})
```
Comment (antoniotrkdz) [inline react-ui/src/actions/login.js]: Done in latest commit.
Much nicer, thank you!

File: react-ui/src/components/Form_Register_Resident.js [hunk 52]
```diff
@@ -1,41 +1,59 @@
...
+
     onSubmit(evt) {
         evt.preventDefault();
-        this.props.registerResident(this.state.resident);
+        var resident = this.state.resident;
+        if (!resident.email) {
+            var error_message = 'Email cannot be empty';
+        }
+        this.setState({errorMessage: error_message}, () => {
+            if (!this.state.errorMessage) {
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Resident.js]: Please ignore it for now...

File: server/index.js (hunk 4 above)
Comment (antoniotrkdz) [inline server/index.js]: Done.

File: server/index.js (hunk 3 above)
Comment (antoniotrkdz) [inline server/index.js]: Checked and corrected.

File: server/index.js (hunk 4 above)
Comment (antoniotrkdz) [inline server/index.js]: Checked and corrected.

File: react-ui/src/components/Login_Form.js [hunk 53]
```diff
@@ -0,0 +1,101 @@
...
+        };
+    }
+
+    onSubmit(evt) {
+        evt.preventDefault();
+        var state = this.props.state;
+        this.setState({errorMessage: ''}, () => {
+            this.props.loginRequest(this.state.email, this.state.password);
+        });
+        (err) => this.setState({
```
Comment (antoniotrkdz) [inline react-ui/src/components/Login_Form.js]: please leave it there I wil fix it when I'll manage the state better.

**Comment by antoniotrkdz** [comment]
Hi, I reviewed all the responses and they are leakproof now.
I also reviewed the naming adopting camelCase for js vars and leaving snake_case for SQL vars.
The things that are not security issue, but just bad code, some I dealt with, some I didn't...please leave them there I will dela with them when I will approach the state management more seriously.

File: react-ui/src/components/Form_Post_Job.js [hunk 54]
```diff
@@ -33,33 +33,34 @@ class Form_Post_Job extends Component {
...
+            //error_message = 'Rate cannot be empty';
+        //} else if (isNaN(jobData.rate) === true) {
+            //error_message = 'Rate must be a number';
+        //} else if (!jobData.description) {
+            //error_message = 'Job Description cannot be empty';
+        //}
 
-        this.setState({errorMessage: error_message}, () => {
-            if (!this.state.errorMessage) {
+        this.setState(//{errorMessage: error_message},
```
Comment (des-des) [inline react-ui/src/components/Form_Post_Job.js]: Fix this.

File: react-ui/src/components/Form_Register_Resident.js (hunk 52 above)
Comment (des-des) [inline react-ui/src/components/Form_Register_Resident.js]:
```js
this.setState({errorMessage: error_message})
if (!error_message) {
  this.props.registerResident(this.state.resident)
}
```

File: react-ui/src/components/Form_Register_Student.js [hunk 55]
```diff
@@ -1,87 +1,116 @@
...
+
     onSubmit(evt) {
         evt.preventDefault();
-        this.props.registerStudent(this.state.student);
+        var student = this.state.student;
+        if (!student.email) {
+            var error_message = 'Email cannot be empty';
+        }
+        this.setState({errorMessage: error_message}, () => {
+            if (!this.state.errorMessage) {
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: Same as above

File: react-ui/src/components/Form_Register_Student.js [hunk 56]
```diff
@@ -1,38 +1,80 @@
...
+        //});
+        //console.log('job-cat', catArray);
+        //Object.assign({}, student, {job_cat:catArray})
+        this.setState({errorMessage: error_message}, () => {
+            if (!this.state.errorMessage) {
+                this.props.registerStudent(student);
+                //browserHistory.push('/dashboard');
+            }
+        });
+        (err) => {
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: Please delete this

File: react-ui/src/components/Form_Register_Student.js (hunk 42 above)
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: please delete this

File: react-ui/src/components/Form_Register_Student.js (hunk 44 above)
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: okay

File: react-ui/src/components/Login_Form.js (hunk 53 above)
Comment (des-des) [inline react-ui/src/components/Login_Form.js]: delete this

File: react-ui/src/components/Login_Form.js (hunk 48 above)
Comment (des-des) [inline react-ui/src/components/Login_Form.js]: delete this

File: react-ui/src/components/Login_Form.js (hunk 49 above)
Comment (des-des) [inline react-ui/src/components/Login_Form.js]: delete this

File: react-ui/src/components/Login_Form.js (hunk 33 above)
Comment (des-des) [inline react-ui/src/components/Login_Form.js]: delete this

File: react-ui/src/components/pages/LoginPage.js [hunk 57]
```diff
@@ -1,58 +1,29 @@
...
-      .then( ({results: items}) => this.setState({items}))
-    }*/
+import * as actions from '../../actions/login.js';
 
+class LoginPage extends React.Component {
     render() {
-        const dispatch = this.props.dispatch;
-        const {formState, currentlySending} = this.props.data;
+        this.props.loginRequest.error;
+        console.log(this.props.loginRequest);
```
Comment (des-des) [inline react-ui/src/components/pages/LoginPage.js]: delete this

File: server/index.js [hunk 58]
```diff
@@ -44,69 +46,166 @@ server.register(plugins, err => {
...
-                        'Failed to connect load data from the database'
-                    );
-                else {
-                    reply({
-                        name: 'jobsList',
-                        message: 'Welcome to BEEVR!',
-                        jobsList: res
-                    });
+                if (err) {
+                    return reply(Boom.serverUnavailable('unavailable: ' + err));
```
Comment (des-des) [inline server/index.js]: maybe dont directly expose db errors.

But not a big problem

File: server/index.js [hunk 59]
```diff
@@ -44,69 +46,166 @@ server.register(plugins, err => {
...
                 }
+                reply({
+                    name: 'jobsList',
+                    message: 'Welcome to BEEVR!',
+                    jobsList: res.map(key => {
```
Comment (des-des) [inline server/index.js]: key is not a key

File: server/index.js [hunk 60]
```diff
@@ -44,69 +46,166 @@ server.register(plugins, err => {
...
+        },
+    });
+
+    server.route({
+        method: 'POST',
+        path: '/api/jobs',
+        handler: (request, reply) => {
+            data.postJobs(request.payload, (err, res) => {
+                if (err) {
+                    return reply(Boom.badRequest('Bad request: ' + err));
```
Comment (des-des) [inline server/index.js]: same but okay

File: server/index.js [hunk 61]
```diff
@@ -44,69 +46,166 @@ server.register(plugins, err => {
...
-                    reply.status(500)(
-                        'Failed to connect load data from the database'
-                    );
+                    return reply(Boom.serverUnavailable('unavailable: ' + err));
                 } else {
                     reply({
                         name: 'studentList',
                         message: 'Welcome to BEEVR!',
-                        studentList: res
+                        studentList: res.map(key => {
```
Comment (des-des) [inline server/index.js]: key is not a key

File: server/index.js [hunk 62]
```diff
@@ -121,14 +220,50 @@ server.register(plugins, err => {
...
+                        message: 'Email sent!',
                     });
                 }
             });
-        }
+        },
+    });
+
+    server.route({
+        method: 'GET',
```
Comment (des-des) [inline server/index.js]: post?

File: server/index.js [hunk 63]
```diff
@@ -138,16 +273,34 @@ server.register(plugins, err => {
...
+                        Boom.serverUnavailable(
+                            'Failed to retrieve data from database'
+                        )
+                    );
                 } else {
                     reply({
                         name: 'myJobsList',
                         message: 'Welcome to BEEVR!',
-                        myJobsList: res
+                        myJobsList: res.map(key => {
```
Comment (des-des) [inline server/index.js]: key is not a key

File: react-ui/src/actions/login.js (hunk 51 above)
Comment (antoniotrkdz) [inline react-ui/src/actions/login.js]: Done for the login too.

File: react-ui/src/components/Form_Register_Student.js (hunk 56 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

File: react-ui/src/components/Form_Register_Student.js (hunk 42 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

File: react-ui/src/components/Login_Form.js (hunk 53 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Login_Form.js]: Done.

File: react-ui/src/components/Login_Form.js (hunk 48 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Login_Form.js]: Done.

File: react-ui/src/components/Login_Form.js (hunk 49 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Login_Form.js]: Done.

File: react-ui/src/actions/login.js [hunk 64]
```diff
@@ -0,0 +1,26 @@
...
```
Comment (antoniotrkdz) [inline react-ui/src/actions/login.js]: Done.

File: react-ui/src/components/Form_Post_Job.js [hunk 65]
```diff
@@ -33,33 +33,34 @@ class Form_Post_Job extends Component {
     _onSubmit(evt) {
         evt.preventDefault();
 
-        var {jobData} = this.state;
+        //var {jobData} = this.state;
```
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Post_Job.js]: Restored.

File: react-ui/src/components/Form_Post_Job.js (hunk 54 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Post_Job.js]: Done.

File: react-ui/src/components/Form_Register_Student.js (hunk 55 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

File: react-ui/src/components/pages/LoginPage.js (hunk 57 above)
Comment (antoniotrkdz) [inline react-ui/src/components/pages/LoginPage.js]: All Done.

File: server/index.js (hunk 59 above)
Comment (antoniotrkdz) [inline server/index.js]: Replaced with element.

File: server/index.js (hunk 62 above)
Comment (antoniotrkdz) [inline server/index.js]: Done.

File: server/index.js (hunk 63 above)
Comment (antoniotrkdz) [inline server/index.js]: Replaced with element.

**Pull request merged**